ui/src/core/app_impl.ts

```diff
@@ -126,6 +126,8 @@ export class AppImpl implements App {
   readonly startupCommandsSetting: Setting<CommandInvocation[]>;
   readonly enforceStartupCommandAllowlistSetting: Setting<boolean>;
   private _isInternalUser?: boolean;
+  private _zenModeEnabled = false;
+  private _topbarVisibleInZenMode = false;
```

Comment:
I think this should not have to be a separate state as this can get out of sync with the omnibox.

I'd just show the topbar whenever the omnibox is not in search mode. Use the existing state rather than trying to add a new bit of state and keep it in sync.